src/mono/mono/mini/aot-compiler.c

```diff
@@ -5352,10 +5352,16 @@ MONO_RESTORE_WARNING
 				for (j = 0; j < decoded_args->named_args_num; ++j) {
 					if (decoded_args->named_args_info [j].field && !strcmp (decoded_args->named_args_info [j].field->name, "EntryPoint")) {
 						named = (const char *)decoded_args->named_args[j]->value.primitive;
-						slen = mono_metadata_decode_value (named, &named) + (int)strlen(acfg->user_symbol_prefix);
-						export_name = (char *)g_malloc (slen + 1);
-						sprintf (export_name, "%s%s", acfg->user_symbol_prefix, named);
```

Comment:
FWIW this could have been modified to
```
namedlen = mono_metadata_decode_value (named, &named);
slen = namedlen + (int)strlen(acfg->user_symbol_prefix);
export_name = (char *)g_malloc (slen + 1);
sprintf (export_name, "%s%.*s", acfg->user_symbol_prefix, namedlen, named);
```

Reply:
Thank you for your suggestion. 
However, even though it achieves the same thing, I prefer not to use `sprintf` as originally it hid the underlying problem that was difficult to track down. So for that reason, I will keep it as is (simple and easily debuggable).